backend/open_webui/routers/files.py

```
@@ -81,20 +86,55 @@ def has_access_to_file(
+    user=Depends(get_verified_user),
 ):
     log.info(f"file.content_type: {file.content_type}")
+
+    if isinstance(metadata, str):
+        try:
+            metadata = json.loads(metadata)
+        except json.JSONDecodeError:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail=ERROR_MESSAGES.DEFAULT("Invalid metadata format"),
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)